scipy/_build_utils/src/complex_blas.c

```diff
@@ -0,0 +1,81 @@
+#include "fortran_defs.h"
+#include "npy_cblas.h"
+#include "npy_2_complexcompat.h"
+
+#ifdef F_SRC
+#include <complex.h>
+#if defined(_MSC_VER) && !defined(__INTEL_COMPILER)
+typedef _Dcomplex double_complex;
+typedef _Fcomplex float_complex;
+#else /* !defined(_MSC_VER) || defined(__INTEL_COMPILER) */
+typedef double _Complex double_complex;
+typedef float _Complex float_complex;
+#endif
+#else
+typedef npy_complex128 double_complex;
+typedef npy_complex64 float_complex;
+#endif
+
+float_complex F_FUNC(wcdotc, WCDOTC)(CBLAS_INT *n, float_complex *cx, CBLAS_INT *incx, float_complex *cy, CBLAS_INT *incy){
+    float_complex ret;
+    CBLAS_FUNC(cblas_cdotc_sub)(*n, cx, *incx, cy, *incy,&ret);
+    return ret;
+}
+
+double_complex F_FUNC(wzdotc, WZDOTC)(CBLAS_INT *n, double_complex *zx, CBLAS_INT *incx, double_complex *zy, CBLAS_INT *incy){
+    double_complex ret;
+    CBLAS_FUNC(cblas_zdotc_sub)(*n, zx, *incx, zy, *incy,&ret);
+    return ret;
+}
+
+float_complex F_FUNC(wcdotu, WCDOTU)(CBLAS_INT *n, float_complex *cx, CBLAS_INT *incx, float_complex *cy, CBLAS_INT *incy){
+    float_complex ret;
+    CBLAS_FUNC(cblas_cdotu_sub)(*n, cx, *incx, cy, *incy,&ret);
+    return ret;
+}
+
+double_complex F_FUNC(wzdotu, WZDOTU)(CBLAS_INT *n, double_complex *zx, CBLAS_INT *incx, double_complex *zy, CBLAS_INT *incy){
+    double_complex ret;
+    CBLAS_FUNC(cblas_zdotu_sub)(*n, zx, *incx, zy, *incy,&ret);
+    return ret;
+}
+
+#ifdef F_SRC
+void BLAS_FUNC(sladiv)(float *xr, float *xi, float *yr, float *yi, float *retr, float * reti);
+void F_FUNC(wcladiv, WCLADIV)(float_complex *ret, float_complex *x, float_complex *y){
+    BLAS_FUNC(sladiv)((float*)(x), (float*)(x)+1, \
+        (float*)(y), (float*)(y)+1, \
+        (float*)(ret), (float*)(ret)+1);
+}
```

Comment:
Is the value of the imaginary components here irrelevant? I don't know `wcladiv`, but just based on the LAPACK [docs](https://netlib.org/lapack/explore-html/d5/db7/group__ladiv.html), just adding `+1` in the imaginary component is not equivalent.

Reply:
C99 `_Complex` types are apparently guaranteed to have the same representation and alignment requirements as an array type containing two elements of the corresponding real type, something I learned when working on this project. That means that this pointer arithmetic should work in theory. However, I just realized that the MSVC complex types do not conform to C99 and are instead structs just like in Cython and Numpy. This makes me wonder whether we would still see segfaults on 32-bit Windows. I don't have any way of testing that, but if it does, I could try declaring the arguments as real arrays of length 2.